src/osd/PeeringState.cc

```diff
@@ -403,17 +407,42 @@ void PeeringState::write_if_dirty(ObjectStore::Transaction& t)
   }
 }
 
+#ifndef WITH_SEASTAR
 void PeeringState::advance_map(
+#else
```

Comment:
i'd suggest have a separated implementation instead of sharing it with the classic OSD. it hurts the readability as the code is segmented into smaller chunks this way.